spec/unit/provider/oneview_san_manager_c7000_spec.rb

```diff
@@ -0,0 +1,137 @@
+      provider.exists?
+      expect { provider.found }.to raise_error(/No SANManager with the specified data were found on the Oneview Appliance/)
+    end
+
+    context 'given the create parameters' do
+      let(:resource) do
+        Puppet::Type.type(:oneview_san_manager).new(
+          name: 'san_manager',
+          ensure: 'present',
+          data:
+              {
+                'providerDisplayName' => 'Brocade Network Advisor',
+                'connectionInfo' => [
+                  {
+                    'name' => 'Host',
+                    'value' => '172.18.15.1'
+                  },
+                  {
+                    'name' => 'Port',
+                    'value' => 5989
+                  },
+                  {
+                    'name' => 'Username',
+                    'value' => 'dcs'
+                  },
+                  {
+                    'name' => 'Password',
+                    'value' => 'dcs'
+                  },
+                  {
+                    'name' => 'UseSsl',
+                    'value' => true
+                  }
+                ]
+              }
+        )
+      end
+
+      it 'should create/add the san manager' do
+        test = resourcetype.new(@client, resource['data'])
+        expect(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([])
+        expect(resourcetype).to receive(:find_by).with(anything, 'providerDisplayName' => resource['data']['providerDisplayName'])
+          .and_return([])
+        provider.exists?
+        allow_any_instance_of(resourcetype).to receive(:add).and_return(test)
+        expect(provider.create).to be
+      end
+
+      it 'should update the san manager' do
+        test = resourcetype.new(@client, resource['data'])
+        expect(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([])
+        expect(resourcetype).to receive(:find_by).with(anything, 'providerDisplayName' => resource['data']['providerDisplayName'])
+          .and_return([test])
+        provider.exists?
+        expect(provider.create).to be
+      end
+
+      it 'should replace provider display name by provider uri' do
+        resource['data']['providerUri'] = 'New Name'
+        test = resourcetype.new(@client, resource['data'])
+        expect(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([test])
+        provider.exists?
+      end
+    end
+
+    context 'given the minimum parameters after San Manager creation' do
+      before(:each) do
+        resource['data']['uri'] = '/rest/san-managers/fake'
+        test = resourcetype.new(@client, resource['data'])
+        allow(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([test])
+        allow(resourcetype).to receive(:get_all).with(anything).and_return([test])
+        provider.exists?
+      end
+      it 'should be able to run through self.instances' do
+        expect(provider).to be
```

Comment:
Aha, I found it weird when I saw this after I just commented it would be able to touch an untested part of the code. 
It should actually be
```ruby
expect(instance).to be
```